perf/perf_watch_point.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2019 IBM
+# Author: Shirisha <shiganta@in..ibm.com>
+
+import os
+import platform
+import shutil
+from avocado import Test
+from avocado import main
+from avocado import skipUnless
+from avocado.utils import cpu, build, distro, process
+from avocado.utils.software_manager import SoftwareManager
+
+CPU_ARCH = cpu.get_cpu_arch().lower()
+IS_POWER8 = 'power8' in CPU_ARCH
+
+
+class PerfWatchPoint(Test):
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.distro_name = detected_distro.name
+        if detected_distro.arch != 'ppc64le':
+            self.cancel('This test is not supported on %s architecture'
+                        % detected_distro.arch)
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in self.distro_name:
+            deps.extend(['linux-tools-common', 'linux-tools-%s' %
+                         platform.uname()[2]])
+        elif self.distro_name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(['perf', 'kernel-devel'])
+        else:
+            self.cancel("Install the package for perf supported \
+                         by %s" % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        shutil.copyfile(self.get_data('swap.S'),
+                        os.path.join(self.teststmpdir, 'swap.S'))
+        shutil.copyfile(self.get_data('wptest.c'),
+                        os.path.join(self.teststmpdir, 'wptest.c'))
+        shutil.copyfile(self.get_data('Makefile'),
+                        os.path.join(self.teststmpdir, 'Makefile'))
+
+        build.make(self.teststmpdir)
+        os.chdir(self.teststmpdir)
+        process.run("insmod wptest.ko")
+
+    def run_cmd(self):
+        i = 1
+        while i <= 4:
+            lst = process.run("cat /dev/wptest")
+            s = list(lst.stdout.strip().split(' '))
```

Comment:
one question is this test case tested against upstream avocado means python3

Reply:
yes praveen i tested
JOB ID     : 4167ca981c3b61adb6e3edc03c9fe912b344a274
JOB LOG    : /root/avocado/job-results/job-2019-09-19T00.08-4167ca9/job.log
 (1/1) perf_watch_point.py:PerfWatchPoint.test_watch_point_check: PASS (3.66 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 4.05 s

Reply:
please have this to work on python3 list(lst.stdout.decode('utf-8').strip().split(' '))